modules/custom/localgov_restricted_content/localgov_restricted_content.module

```diff
@@ -0,0 +1,112 @@
+    $anonymous_view_mode =  $view_mode . '_anonymous';
+    if ($entity_view_display->load('node.' . $entity->bundle() . '.' . $anonymous_view_mode)) {
+      $view_mode = $anonymous_view_mode;
+    }
+  }
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function localgov_restricted_content_form_node_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+
+  /** @var \Drupal\node\NodeForm $node_form */
+  $node_form = $form_state->getBuildInfo()['callback_object'];
+
+  if (!($node_form instanceof NodeForm)) {
+    return;
+  }
+
+  /** @var \Drupal\Core\Entity\ContentEntityInterface $entity */
+  $entity = $node_form->getEntity();
+
+  /** @var \Drupal\localgov_restricted_content\RestrictedContentInterface $restricted_content */
+  $restricted_content = \Drupal::service('localgov_restricted_content.restricted_content');
+
+  if ($restricted_content->isAncestorRestricted($entity)) {
+    $form['localgov_restricted_content']['#suffix'] = t('<div class="form-item__label option">This is disabled because the page\'s parent is restricted.</div>');
+    $form['localgov_restricted_content']['#disabled'] = TRUE;
+  }
+}
+
+/**
+ * Implements hook_entity_extra_field_info().
+ */
+function localgov_restricted_content_entity_extra_field_info() {
+  $return = [];
+  $bundle_info = \Drupal::service('entity_type.bundle.info')->getBundleInfo('node');
+  foreach ($bundle_info as $bundle => $bundle_info) {
+    $return['node'][$bundle]['display']['restricted_content_sign_in'] = [
+      'label' => t('Sign in link'),
+      'description' => t('Display sign in link at the bottom'),
+      'visible' => TRUE,
+    ];
+    $return['node'][$bundle]['display']['restricted_content_sign_in_with_message'] = [
+      'label' => t('Sign in link with message'),
+      'description' => t('Display sign in link with message at the bottom'),
+      'visible' => TRUE,
+    ];
+  }
+
+  return $return;
+}
+
+/**
+ * Implements hook_entity_view().
+ *
+ * Display sign-in message and/or link for anonymous users.
+ */
+function localgov_restricted_content_entity_view(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode) {
+  if (\Drupal::currentUser()->isAuthenticated()) {
+    return;
+  }
+  $content = $display->get('content');
+  if (isset($content['restricted_content_sign_in_with_message'])) {
+    $build['sign_in'] = [
+      '#theme' => 'localgov_restricted_content_sign_in_with_message',
+    ];
+  }
+  elseif (isset($content['restricted_content_sign_in'])) {
+    $build['sign_in'] = [
+      '#theme' => 'localgov_restricted_content_sign_in',
+    ];
+  }
+}
+
+/**
+ * Implements hook_theme().
+ */
+function localgov_restricted_content_theme($existing, $type, $theme, $path) {
+  return [
+    'localgov_restricted_content_sign_in' => [
+      'render element' => 'elements',
+      ],
+    'localgov_restricted_content_sign_in_with_message' => [
+      'render element' => 'elements',
+    ],
+  ];
+}
```

Comment:
Out of curiosity, does Drupal's coding standards no longer require a trailing new line?

Reply:
Codesniffer didn't spot it, don't know why.